src/AbstractOperations/grid_metrics.jl

```diff
@@ -133,11 +135,59 @@ Adapt.adapt_structure(to, gm::GridMetricOperation{LX, LY, LZ}) where {LX, LY, LZ
 on_architecture(to, gm::GridMetricOperation{LX, LY, LZ}) where {LX, LY, LZ} =
     GridMetricOperation{LX, LY, LZ}(on_architecture(to, gm.metric),
                                     on_architecture(to, gm.grid))
-                                
+
 
 @inline Base.getindex(gm::GridMetricOperation, i, j, k) = gm.metric(i, j, k, gm.grid)
 
 indices(::GridMetricOperation) = default_indices(3)
 
 # Special constructor for BinaryOperation
 GridMetricOperation(L, metric, grid) = GridMetricOperation{L[1], L[2], L[3]}(metric_function(L, metric), grid)
+
+#####
+##### Spacings
+#####
+
+function xspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δx_op = KernelFunctionOperation{LX, LY, LZ}(xspacing, grid, ℓx, ℓy, ℓz)
+    return Δx_op
+end
+
+function yspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δy_op = KernelFunctionOperation{LX, LY, LZ}(yspacing, grid, ℓx, ℓy, ℓz)
+    return Δy_op
+end
+
+function zspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δz_op = KernelFunctionOperation{LX, LY, LZ}(zspacing, grid, ℓx, ℓy, ℓz)
+    return Δz_op
+end
+
+function λspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δλ_op = KernelFunctionOperation{LX, LY, LZ}(λspacing, grid, ℓx, ℓy, ℓz)
+    return Δλ_op
+end
+
+function φspacings(grid, ℓx, ℓy, ℓz)
+    LX, LY, LZ = typeof.((ℓx, ℓy, ℓz))
+    Δφ_op = KernelFunctionOperation{LX, LY, LZ}(φspacing, grid, ℓx, ℓy, ℓz)
+    return Δφ_op
+end
+
+@inline xspacings(grid, ℓx) = xspacings(grid, ℓx, Center(), Center())
+@inline yspacings(grid, ℓy) = yspacings(grid, Center(), ℓy, Center())
+@inline zspacings(grid, ℓz) = zspacings(grid, Center(), Center(), ℓz)
+
+@inline λspacings(grid, ℓx) = λspacings(grid, ℓx, Center(), Center())
+@inline φspacings(grid, ℓy) = φspacings(grid, Center(), ℓy, Center())
```

Comment:
These are valid for `LatitudeLongitudeGrid` but not `OrthogonalSphericalShellGrid`

Reply:
These definitions have been moved to the grid-specific files.